client/src/pages/Payment/Payment.tsx

```diff
@@ -0,0 +1,192 @@
+import { useState, useEffect } from 'react';
+import PaymentMethodForm from './PaymentMethodForm/PaymentMethodForm';
+import masterCardLogo from './CardImages/Mastercard-Logo.svg';
+import visaCardLogo from './CardImages/Visacard-Logo.svg';
+import starOfLife from './CardImages/star-of-life.png';
+import {
+  Card,
+  Grid,
+  Typography,
+  Button,
+  DialogTitle,
+  DialogContent,
+  Dialog,
+  CardContent,
+  Checkbox,
+} from '@material-ui/core';
+import useStyles from './useStyles';
+import { useAuth } from '../../context/useAuthContext';
+import { retrievePaymentProfiles, getUserStripeId, setDefaultPaymentProfile } from '../../helpers/APICalls/payment';
+import { usePaymentProfiles } from '../../context/usePaymentProfilesContext';
+
+export default function PaymentProfile(): JSX.Element {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  const { loggedInUser } = useAuth();
+  const { updatePaymentProfiles, paymentProfiles, updateLoggedInUser } = usePaymentProfiles();
+  const [paymentProfileExist, SetPaymentProfileExist] = useState<boolean>(false);
+  const [userIds, setUserIds] = useState<any>(false);
```

Comment:
Please avoid using 'any' as a type

Ditto throughout!

Reply:
I will make changes but please how can I declare the data type of a promise containing an object with lots of Properties?

Reply:
You can define an interface and use it 👍🏽